pkg/smokescreen/metrics_test.go

```diff
@@ -56,3 +61,92 @@ func TestMetricsClient(t *testing.T) {
 		r.Error(err)
 	})
 }
+
+// MockMetricsClient is a MetricsClient that counts metric updates.
```

Comment:
I found it surprising that this was defined here, but only used in smokescreen_test.go--have you thought about putting it there instead?

Reply:
As of c4e1bb7, `metrics_test.go` now uses this machinery too. :-)